edgelet/edgelet-core/src/settings.rs

```diff
@@ -411,7 +418,15 @@ impl Provisioning {
     }
 
     pub fn dynamic_reprovisioning(&self) -> bool {
-        self.dynamic_reprovisioning
+        // If using DPS provisioning and its always_reprovision_on_startup is false,
+        // then dynamic reprovisioning is the only way we can know when the device identity has been deleted in IoT Hub.
+        //
+        // So force dynamic_reprovisioning to be true even if the user didn't set it.
+
+        match &self.provisioning {
```

Comment:
As we discussed, I've removed the override entirely so that the user can keep it turned off in case there's a bug that triggers spurious reprovisioning.